Add the missing -y option to postgres install line

khashf opened the pull request:
Tested on my instance. The script stopped midway and asked for my confirmation for postgresql yum install, which it shouldn't.

MikeTheCanuck commented:
This is a great catch and a desirable PR.

Can I ask about exactly what environment you ran this in that it stopped for confirmation?  I ran this script a few dozen times on live production EC2 machines and I don't remember the last time it coughed at this step.

My scenario:
- Create an EC2 instance - the "Amazon Linux 2 LTS Candidate AMI 2017.12.0 (HVM), SSD Volume Type - ami-7f43f307"
- instance type = `t2.micro`
- instance details = default
- add storage = [none]
- added a single `Name` tag
- used an existing security group that allows tcp/22 and tcp/5432
- Finally, I `scp`'d the script to the new instance, `ssh`'d as `ec2-user` and ran the script immediately as `./create-database-development.sh`

khashf commented:
My specs:

- Image: Amazon Linux 2 LTS Candidate (ami-7f43f307)
- Instance type: t2.micro
- SSD Volume Type
- Instance detail: default
- Add storage: I added an EBS volume to mount at /data later on
- Name: a "Name" tag with value read from $1 of my wrapper script
- Security Group: Amazon's default security group. Allow SSH inbound and all outbound

After run my script below:

1. I ssh into the instance, `wget` the [link the raw version of the script](https://raw.githubusercontent.com/hackoregon/civic-devops/master/bin/create-database-development.sh)
2. `chmod +x` `the script`
3. run the script with `sudo`

Here is my wrapping script for `aws ec2 run-instances`
```
instance_name=$1
tag_specs='ResourceType=instance,Tags=[{Key=Name,Value='$instance_name'}]'
echo $tag_specs
aws ec2 run-instances \
    --image-id ami-7f43f307 \
    --count 1 \
    --instance-type t2.micro \
    --key-name <my-key-name> \
    --security-group-ids <aws vpc's default security group id> \
    --subnet-id <subnet within the vpc> \
    --region us-west-2 \
    --block-device-mappings "[{\"DeviceName\":\"/dev/sdf\",\"Ebs\":{\"VolumeSize\":8,\"DeleteOnTermination\":true}}]" \
    --tag-specifications $tag_specs
```

Here are the commands I issued and the outputs I got. 
As you can see, the script stops midway and ask for confirmation. 
Putting -y option fixes this issue.
```

       __|  __|_  )
       _|  (     /   Amazon Linux 2 AMI
      ___|\___|___|

https://aws.amazon.com/amazon-linux-2/
No packages needed for security; 15 packages available
Run "sudo yum update" to apply all updates.
**[ec2-user@ip-172-31-30-27 ~]$** wget https://raw.githubusercontent.com/hackoregon/civic-devops/master/bin/create-database-development.sh
--2018-03-05 07:42:05--  https://raw.githubusercontent.com/hackoregon/civic-devops/master/bin/create-database-development.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.52.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.52.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5400 (5.3K) [text/plain]
Saving to: ‘create-database-development.sh’

100%[==================================================================================================================================================================>] 5,400       --.-K/s   in 0s      

2018-03-05 07:42:05 (92.3 MB/s) - ‘create-database-development.sh’ saved [5400/5400]

**[ec2-user@ip-172-31-30-27 ~]$** chmod +x ./create-database-development.sh 
**[ec2-user@ip-172-31-30-27 ~]$** sudo ./create-database-development.sh 
Installing PostgreSQL packages...
Loaded plugins: langpacks, priorities, update-motd
Resolving Dependencies
--> Running transaction check
---> Package bind-libs.x86_64 32:9.9.4-51.amzn2 will be updated
---> Package bind-libs.x86_64 32:9.9.4-51.amzn2.2 will be an update
---> Package bind-libs-lite.x86_64 32:9.9.4-51.amzn2 will be updated
---> Package bind-libs-lite.x86_64 32:9.9.4-51.amzn2.2 will be an update
---> Package bind-license.noarch 32:9.9.4-51.amzn2 will be updated
---> Package bind-license.noarch 32:9.9.4-51.amzn2.2 will be an update
---> Package bind-utils.x86_64 32:9.9.4-51.amzn2 will be updated
---> Package bind-utils.x86_64 32:9.9.4-51.amzn2.2 will be an update
---> Package curl.x86_64 0:7.55.1-8.amzn2.0.1 will be updated
---> Package curl.x86_64 0:7.55.1-9.amzn2.0.1 will be an update
---> Package dhclient.x86_64 12:4.2.5-58.amzn2.0.1 will be updated
---> Package dhclient.x86_64 12:4.2.5-58.amzn2.1.2 will be an update
---> Package dhcp-common.x86_64 12:4.2.5-58.amzn2.0.1 will be updated
---> Package dhcp-common.x86_64 12:4.2.5-58.amzn2.1.2 will be an update
---> Package dhcp-libs.x86_64 12:4.2.5-58.amzn2.0.1 will be updated
---> Package dhcp-libs.x86_64 12:4.2.5-58.amzn2.1.2 will be an update
---> Package kernel.x86_64 0:4.9.81-44.57.amzn2 will be installed
---> Package kernel-tools.x86_64 0:4.9.76-38.79.amzn2 will be updated
---> Package kernel-tools.x86_64 0:4.9.81-44.57.amzn2 will be an update
---> Package libcurl.x86_64 0:7.55.1-8.amzn2.0.1 will be updated
---> Package libcurl.x86_64 0:7.55.1-9.amzn2.0.1 will be an update
---> Package system-release.x86_64 0:2017.12-1.amzn2 will be updated
---> Package system-release.x86_64 0:2017.12-2.amzn2 will be an update
---> Package systemd.x86_64 0:219-42.amzn2.4 will be updated
---> Package systemd.x86_64 0:219-42.amzn2.7 will be an update
---> Package systemd-libs.x86_64 0:219-42.amzn2.4 will be updated
---> Package systemd-libs.x86_64 0:219-42.amzn2.7 will be an update
---> Package systemd-sysv.x86_64 0:219-42.amzn2.4 will be updated
---> Package systemd-sysv.x86_64 0:219-42.amzn2.7 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

============================================================================================================================================================================================================
 Package                                           Arch                                      Version                                                    Repository                                     Size
============================================================================================================================================================================================================
Installing:
 kernel                                            x86_64                                    4.9.81-44.57.amzn2                                         amzn2-core                                     17 M
Updating:
 bind-libs                                         x86_64                                    32:9.9.4-51.amzn2.2                                        amzn2-core                                    1.0 M
 bind-libs-lite                                    x86_64                                    32:9.9.4-51.amzn2.2                                        amzn2-core                                    734 k
 bind-license                                      noarch                                    32:9.9.4-51.amzn2.2                                        amzn2-core                                     84 k
 bind-utils                                        x86_64                                    32:9.9.4-51.amzn2.2                                        amzn2-core                                    203 k
 curl                                              x86_64                                    7.55.1-9.amzn2.0.1                                         amzn2-core                                    304 k
 dhclient                                          x86_64                                    12:4.2.5-58.amzn2.1.2                                      amzn2-core                                    285 k
 dhcp-common                                       x86_64                                    12:4.2.5-58.amzn2.1.2                                      amzn2-core                                    174 k
 dhcp-libs                                         x86_64                                    12:4.2.5-58.amzn2.1.2                                      amzn2-core                                    130 k
 kernel-tools                                      x86_64                                    4.9.81-44.57.amzn2                                         amzn2-core                                     93 k
 libcurl                                           x86_64                                    7.55.1-9.amzn2.0.1                                         amzn2-core                                    269 k
 system-release                                    x86_64                                    2017.12-2.amzn2                                            amzn2-core                                     17 k
 systemd                                           x86_64                                    219-42.amzn2.7                                             amzn2-core                                    5.1 M
 systemd-libs                                      x86_64                                    219-42.amzn2.7                                             amzn2-core                                    380 k
 systemd-sysv                                      x86_64                                    219-42.amzn2.7                                             amzn2-core                                     71 k

Transaction Summary
============================================================================================================================================================================================================
Install   1 Package
Upgrade  14 Packages

Total download size: 25 M
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
(1/15): bind-libs-9.9.4-51.amzn2.2.x86_64.rpm                                                                                                                                        | 1.0 MB  00:00:00     
(2/15): bind-libs-lite-9.9.4-51.amzn2.2.x86_64.rpm                                                                                                                                   | 734 kB  00:00:00     
(3/15): bind-license-9.9.4-51.amzn2.2.noarch.rpm                                                                                                                                     |  84 kB  00:00:00     
(4/15): curl-7.55.1-9.amzn2.0.1.x86_64.rpm                                                                                                                                           | 304 kB  00:00:00     
(5/15): bind-utils-9.9.4-51.amzn2.2.x86_64.rpm                                                                                                                                       | 203 kB  00:00:00     
(6/15): dhclient-4.2.5-58.amzn2.1.2.x86_64.rpm                                                                                                                                       | 285 kB  00:00:00     
(7/15): dhcp-libs-4.2.5-58.amzn2.1.2.x86_64.rpm                                                                                                                                      | 130 kB  00:00:00     
(8/15): dhcp-common-4.2.5-58.amzn2.1.2.x86_64.rpm                                                                                                                                    | 174 kB  00:00:00     
(9/15): kernel-tools-4.9.81-44.57.amzn2.x86_64.rpm                                                                                                                                   |  93 kB  00:00:00     
(10/15): libcurl-7.55.1-9.amzn2.0.1.x86_64.rpm                                                                                                                                       | 269 kB  00:00:00     
(11/15): system-release-2017.12-2.amzn2.x86_64.rpm                                                                                                                                   |  17 kB  00:00:00     
(12/15): systemd-219-42.amzn2.7.x86_64.rpm                                                                                                                                           | 5.1 MB  00:00:00     
(13/15): kernel-4.9.81-44.57.amzn2.x86_64.rpm                                                                                                                                        |  17 MB  00:00:00     
(14/15): systemd-sysv-219-42.amzn2.7.x86_64.rpm                                                                                                                                      |  71 kB  00:00:00     
(15/15): systemd-libs-219-42.amzn2.7.x86_64.rpm                                                                                                                                      | 380 kB  00:00:00     
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                                                        15 MB/s |  25 MB  00:00:01     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : systemd-libs-219-42.amzn2.7.x86_64                                                                                                                                                      1/29 
  Updating   : 12:dhcp-libs-4.2.5-58.amzn2.1.2.x86_64                                                                                                                                                  2/29 
  Updating   : 32:bind-license-9.9.4-51.amzn2.2.noarch                                                                                                                                                 3/29 
  Updating   : libcurl-7.55.1-9.amzn2.0.1.x86_64                                                                                                                                                       4/29 
  Updating   : 32:bind-libs-lite-9.9.4-51.amzn2.2.x86_64                                                                                                                                               5/29 
  Updating   : 32:bind-libs-9.9.4-51.amzn2.2.x86_64                                                                                                                                                    6/29 
  Updating   : 12:dhcp-common-4.2.5-58.amzn2.1.2.x86_64                                                                                                                                                7/29 
  Updating   : system-release-2017.12-2.amzn2.x86_64                                                                                                                                                   8/29 
  Updating   : systemd-219-42.amzn2.7.x86_64                                                                                                                                                           9/29 
  Updating   : systemd-sysv-219-42.amzn2.7.x86_64                                                                                                                                                     10/29 
  Updating   : 12:dhclient-4.2.5-58.amzn2.1.2.x86_64                                                                                                                                                  11/29 
  Updating   : 32:bind-utils-9.9.4-51.amzn2.2.x86_64                                                                                                                                                  12/29 
  Updating   : curl-7.55.1-9.amzn2.0.1.x86_64                                                                                                                                                         13/29 
  Updating   : kernel-tools-4.9.81-44.57.amzn2.x86_64                                                                                                                                                 14/29 
  Installing : kernel-4.9.81-44.57.amzn2.x86_64                                                                                                                                                       15/29 
  Cleanup    : 12:dhclient-4.2.5-58.amzn2.0.1.x86_64                                                                                                                                                  16/29 
  Cleanup    : 12:dhcp-common-4.2.5-58.amzn2.0.1.x86_64                                                                                                                                               17/29 
  Cleanup    : 12:dhcp-libs-4.2.5-58.amzn2.0.1.x86_64                                                                                                                                                 18/29 
  Cleanup    : systemd-sysv-219-42.amzn2.4.x86_64                                                                                                                                                     19/29 
  Cleanup    : systemd-219-42.amzn2.4.x86_64                                                                                                                                                          20/29 
  Cleanup    : 32:bind-libs-lite-9.9.4-51.amzn2.x86_64                                                                                                                                                21/29 
  Cleanup    : 32:bind-utils-9.9.4-51.amzn2.x86_64                                                                                                                                                    22/29 
  Cleanup    : 32:bind-libs-9.9.4-51.amzn2.x86_64                                                                                                                                                     23/29 
  Cleanup    : curl-7.55.1-8.amzn2.0.1.x86_64                                                                                                                                                         24/29 
  Cleanup    : 32:bind-license-9.9.4-51.amzn2.noarch                                                                                                                                                  25/29 
  Cleanup    : system-release-2017.12-1.amzn2.x86_64                                                                                                                                                  26/29 
  Cleanup    : libcurl-7.55.1-8.amzn2.0.1.x86_64                                                                                                                                                      27/29 
  Cleanup    : systemd-libs-219-42.amzn2.4.x86_64                                                                                                                                                     28/29 
  Cleanup    : kernel-tools-4.9.76-38.79.amzn2.x86_64                                                                                                                                                 29/29 
  Verifying  : 32:bind-libs-lite-9.9.4-51.amzn2.2.x86_64                                                                                                                                               1/29 
  Verifying  : 12:dhclient-4.2.5-58.amzn2.1.2.x86_64                                                                                                                                                   2/29 
  Verifying  : libcurl-7.55.1-9.amzn2.0.1.x86_64                                                                                                                                                       3/29 
  Verifying  : systemd-219-42.amzn2.7.x86_64                                                                                                                                                           4/29 
  Verifying  : curl-7.55.1-9.amzn2.0.1.x86_64                                                                                                                                                          5/29 
  Verifying  : systemd-libs-219-42.amzn2.7.x86_64                                                                                                                                                      6/29 
  Verifying  : 12:dhcp-libs-4.2.5-58.amzn2.1.2.x86_64                                                                                                                                                  7/29 
  Verifying  : 32:bind-libs-9.9.4-51.amzn2.2.x86_64                                                                                                                                                    8/29 
  Verifying  : system-release-2017.12-2.amzn2.x86_64                                                                                                                                                   9/29 
  Verifying  : 12:dhcp-common-4.2.5-58.amzn2.1.2.x86_64                                                                                                                                               10/29 
  Verifying  : kernel-4.9.81-44.57.amzn2.x86_64                                                                                                                                                       11/29 
  Verifying  : systemd-sysv-219-42.amzn2.7.x86_64                                                                                                                                                     12/29 
  Verifying  : kernel-tools-4.9.81-44.57.amzn2.x86_64                                                                                                                                                 13/29 
  Verifying  : 32:bind-utils-9.9.4-51.amzn2.2.x86_64                                                                                                                                                  14/29 
  Verifying  : 32:bind-license-9.9.4-51.amzn2.2.noarch                                                                                                                                                15/29 
  Verifying  : 12:dhcp-common-4.2.5-58.amzn2.0.1.x86_64                                                                                                                                               16/29 
  Verifying  : 12:dhclient-4.2.5-58.amzn2.0.1.x86_64                                                                                                                                                  17/29 
  Verifying  : 32:bind-libs-lite-9.9.4-51.amzn2.x86_64                                                                                                                                                18/29 
  Verifying  : 32:bind-utils-9.9.4-51.amzn2.x86_64                                                                                                                                                    19/29 
  Verifying  : system-release-2017.12-1.amzn2.x86_64                                                                                                                                                  20/29 
  Verifying  : 32:bind-libs-9.9.4-51.amzn2.x86_64                                                                                                                                                     21/29 
  Verifying  : systemd-sysv-219-42.amzn2.4.x86_64                                                                                                                                                     22/29 
  Verifying  : libcurl-7.55.1-8.amzn2.0.1.x86_64                                                                                                                                                      23/29 
  Verifying  : systemd-219-42.amzn2.4.x86_64                                                                                                                                                          24/29 
  Verifying  : curl-7.55.1-8.amzn2.0.1.x86_64                                                                                                                                                         25/29 
  Verifying  : systemd-libs-219-42.amzn2.4.x86_64                                                                                                                                                     26/29 
  Verifying  : kernel-tools-4.9.76-38.79.amzn2.x86_64                                                                                                                                                 27/29 
  Verifying  : 12:dhcp-libs-4.2.5-58.amzn2.0.1.x86_64                                                                                                                                                 28/29 
  Verifying  : 32:bind-license-9.9.4-51.amzn2.noarch                                                                                                                                                  29/29 

Installed:
  kernel.x86_64 0:4.9.81-44.57.amzn2                                                                                                                                                                        

Updated:
  bind-libs.x86_64 32:9.9.4-51.amzn2.2    bind-libs-lite.x86_64 32:9.9.4-51.amzn2.2 bind-license.noarch 32:9.9.4-51.amzn2.2 bind-utils.x86_64 32:9.9.4-51.amzn2.2    curl.x86_64 0:7.55.1-9.amzn2.0.1   
  dhclient.x86_64 12:4.2.5-58.amzn2.1.2   dhcp-common.x86_64 12:4.2.5-58.amzn2.1.2  dhcp-libs.x86_64 12:4.2.5-58.amzn2.1.2  kernel-tools.x86_64 0:4.9.81-44.57.amzn2 libcurl.x86_64 0:7.55.1-9.amzn2.0.1
  system-release.x86_64 0:2017.12-2.amzn2 systemd.x86_64 0:219-42.amzn2.7           systemd-libs.x86_64 0:219-42.amzn2.7    systemd-sysv.x86_64 0:219-42.amzn2.7    

Complete!
  0  ansible2   disabled  [ =2.4.2 ]
  1  emacs   disabled  [ =25.3 ]
  2  memcached1.5   disabled  [ =1.5.1 ]
  3  nginx1.12   disabled  [ =1.12.2 ]
  4  postgresql9.6=latest  enabled  [ =9.6.6 ]
  5  python3   disabled  [ =3.6.2 ]
  6  redis4.0   disabled  [ =4.0.5 ]
  7  R3.4   disabled  [ =3.4.3 ]
  8  rust1   disabled  [ =1.22.1 ]
  9  vim   disabled  [ =8.0 ]
 10  golang1.9   disabled  [ =1.9.2 ]
 11  ruby2.4   disabled  [ =2.4.2 ]
 12  nano   disabled  [ =2.9.1 ]
 13  php7.2   disabled  [ =7.2.0 ]
 14  lamp-mariadb10.2-php7.2   disabled  [ =10.2.10_7.2.0 ]
Loaded plugins: langpacks, priorities, update-motd
Package postgresql-9.6.6-1.amzn2.0.1.x86_64 already installed and latest version
Resolving Dependencies
--> Running transaction check
---> Package postgresql-server.x86_64 0:9.6.6-1.amzn2.0.1 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

============================================================================================================================================================================================================
 Package                                           Arch                                   Version                                            Repository                                                Size
============================================================================================================================================================================================================
Installing:
 postgresql-server                                 x86_64                                 9.6.6-1.amzn2.0.1                                  amzn2extra-postgresql9.6                                 4.6 M

Transaction Summary
============================================================================================================================================================================================================
Install  1 Package

Total download size: 4.6 M
Installed size: 19 M
Is this ok [y/d/N]: 

```

MikeTheCanuck commented:
How embarrassing - I ran this so often I completely ignored it.  Good catch.
